spec/registry.md

```diff
@@ -619,7 +619,43 @@ All other values produce an _Invalid Expression_ error.
 > or the type `com.ibm.icu.util.CurrencyAmount` can be used to set the currency and related
 > options (such as the number of fraction digits).
 
+### Non-Negative Integer Options
 
+Some _options_ of number _functions_ are defined to take a "non-negative integer".
+Implementations of number _functions_ use these _options_ to control aspects of numeric display
+such as the number of fraction, integer, or significant digits.
+
+A "non-negative integer" is an _option_ value that the _function_ evaluates as an integer 
+greater than or equal to zero.
+Implementations MAY define an upper limit on the resolved value 
+of a non-negative integer option consistent with that implementation's practical limits.
+
+The implementation MAY accept any implementation-defined type as the value.
+Implementations MUST accept the value as a _literal_.
+When the value is a _literal_,
+it MUST consist of a sequence of ASCII digits in the range U+0030 through U+0039.
```

Comment:
The MUST for literals + MAY for variables is _fine_, though it feels a bit clumsy -- I don't really see the need for allowing implementations to vary here.

Beyond that, I do have one remaining qualm about this: There's no reason for us to explicitly allow for e.g. `001` to be a valid representation of an integer. We've spec'd `number-literal` to match JSON, and if we don't restrict this to be a subset of its number representation, then an implementation could not use a JSON parser to parse and validate all numeric values.

In ABNF, the matching rule is `%x30 / (%x31-39 *DIGIT)`, which probably could be used here as well.

```suggestion
When the value is a _literal_,
it MUST consist of a sequence of ASCII digits matching the regular expression `^0|[1-9][0-9]*$`.
```

Reply:
I didn't rule out zero prefixes (and I show one in the example set below) but we could do that here. We haven't used much with regex elsewhere, so we might want to stick to ABNF here also.

```suggestion
When the value is a _literal_, it MUST match `non-negative` in this ABNF:
>```abnf
>non-negative = "0" / (("1"-"9") *DIGIT)
>```
```